src/dotnet/UI.Blazor/Components/ContextMenu/BlazorContextMenu/Services/BlazorContextMenuService.cs

```diff
@@ -0,0 +1,70 @@
+using BlazorContextMenu.Services;
+using Microsoft.JSInterop;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace BlazorContextMenu
+{
+    public interface IBlazorContextMenuService
+    {
+        /// <summary>
+        /// Hides a <see cref="ContextMenu" /> programmatically.
+        /// </summary>
+        /// <param name="id"></param>
+        /// <returns></returns>
+        Task HideMenu(string id);
+
+        /// <summary>
+        /// Shows a <see cref="ContextMenu" /> programmatically.
+        /// </summary>
+        /// <param name="id">The id of the <see cref="ContextMenu"/>.</param>
+        /// <param name="x"></param>
+        /// <param name="y"></param>
+        /// <returns></returns>
+        Task ShowMenu(string id, int x, int y);
+
+        /// <summary>
+        /// Shows a <see cref="ContextMenu" /> programmatically.
+        /// </summary>
+        /// <param name="id">The id of the <see cref="ContextMenu"/>.</param>
+        /// <param name="x"></param>
+        /// <param name="y"></param>
+        /// <param name="data">Extra data that will be passed to menu events.</param>
+        /// <returns></returns>
+        Task ShowMenu(string id, int x, int y, object? data);
+    }
+
+    public class BlazorContextMenuService : IBlazorContextMenuService
+    {
+        private readonly IJSRuntime _jSRuntime;
```

Comment:
1. We've agreed to use properties for all injected members.
2. We call `IJSRuntime` member as `JS`.

Reply:
On p.1, please fix it everywhere in this changeset.

Reply:
done